### PR TITLE
SL-1708 add support for TrueAdapter

### PR DESCRIPTION
**Pull request by rlvrs**

### CODE REVIEW

File: lib/feature_flags/adapters/null_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# typed: strict
+# frozen_string_literal: true
+
+module FeatureFlags
+  module Adapters
+    class NullAdapter < Adapter
+      extend T::Sig
+
+      sig { override.params(account_request: Params::AccountRequest).returns(T::Boolean) }
+      def feature_account_enabled?(account_request)
```
Comment (tjmateus) [inline lib/feature_flags/adapters/null_adapter.rb:10]: Is it supposed to have the next two lines?

File: lib/feature_flags/adapters/null_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        account_request.feature.empty?
```
Comment (tjmateus) [inline lib/feature_flags/adapters/null_adapter.rb:11]: Is it supposed to have this line if we want to return always true?

**Comment by dmbento** [comment]
IMO: Null adapter is rather strange in this case (sounds like there's no implementations for it but there is on fact). I'd change this to a `ConstantAdapter` that allows to define the output as `true` or `false`, or create a `TruthAdapter` and the opposite one.

**Review by migueltaoliveira — APPROVED** [review]

**Comment by migueltaoliveira** [comment]
> IMO: Null adapter is rather strange in this case (sounds like there's no implementations for it but there is on fact). I'd change this to a `ConstantAdapter` that allows to define the output as `true` or `false`, or create a `TruthAdapter` and the opposite one.

We are already using the NullStrategy's on the platform when we want to bypass the request, it's like a 'do nothing adapter'. This is a design pattern: https://sourcemaking.com/design_patterns/null_object: 
> Since a null object usually does not have any state, its state can't change

File: lib/feature_flags/adapters/null_adapter.rb (hunk 2 above)
Comment (rlvrs) [reply lib/feature_flags/adapters/null_adapter.rb:11]: whoopsy

**Comment by dmbento** [comment]
@migueltaoliveira the problem is that is does something, it evaluates to true, and that's a valid strategy.

**Comment by migueltaoliveira** [comment]
> @migueltaoliveira the problem is that is does something, it evaluates to true, and that's a valid strategy.

But the state does not change, so it is a null implementation.

**Comment by dmbento** [comment]
what if you want another adapter that resolves to `false`? what would you name it?

**Comment by migueltaoliveira** [comment]
> what if you want another adapter that resolves to `false`? what would you name it?

If you want to resolve to false, the name `FalseAdapter` makes sense, since you're not bypassing anymore. Anyway, I believe that `TrueAdapter` it's easy to understand, so it's a good name as well :)

**Pull request merged**